src/app/contribute/guidelines/component-guidelines/index.html

```diff
@@ -0,0 +1,310 @@
+<stache
+  pageTitle="Component guidelines"
+  showTableOfContents="true"
+  layout="sidebar">
+
+  <stache-page-summary>
+    <p>
+      We highly encourage contributions from all SKY UX users, and we ask that you follow these general guidelines so we can review and accept your contributions in a timely fashion. If you do not follow these guidelines, we cannot accept your contributions.
+    </p>
+  </stache-page-summary>
+
+  <stache-page-anchor>
+    Prerequisites
+  </stache-page-anchor>
+
+  <p>Before you contribute, you must install the <a stacheRouterLink="/learn/get-started/prereqs">SKY UX prerequisites</a>.</p>
+
+  <stache-page-anchor>
+    Contribution guidelines
+  </stache-page-anchor>
+
+  <ul>
+    <li>
+      <b>File an issue before you start any work.</b> The sooner we can discuss potential code changes, the better. We can also use the issue to track and assign any work needed. For guidance, see <a stacheRouterLink="." fragment="file-an-issue">file an issue</a>.
+    </li>
+    <li>
+      <b>Use the <stache-code>Sky</stache-code> prefix when naming all classes, directives, services, components, etc.</b> This prefix indicates to other contributors that items are owned by SKY UX and not a third-party library and also prevents potential class-name clashes with other libraries. Keep in mind that while we generally use the uppercase <stache-code>Sky</stache-code> prefix, we also use the <stache-code>sky-</stache-code> prefix in some cases, such as the selector property in components.
+    </li>
+    <li>
+      <b>All new code must have 100% unit test code coverage.</b> This doesn't guarantee that every use case is accounted for, but anything less than 100% code coverage does guarantee that at least one use case is not accounted for. This can be verified by running tests with <stache-code>npm run watch</stache-code> and viewing the code coverage results in <stache-code>coverage/Chrome xx.x.xxxx/index.html</stache-code>.
+    </li>
+    <li>
+      <b>All new components and visual changes to existing components must be accompanied by visual regression tests.</b> This ensures that future changes to CSS or markup will not cause components to render in an unexpected manner. Visual tests consist of three parts: an HTML template for the component to test, a TypeScript file for the component to test, and the actual file that runs the test using webdriver.io and our custom screenshot functions. You can see examples of each part of the visual test process at:
+    </li>
+    <ul>
+      <li>
+        <a stacheRouterLink="https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.html">https://github.com/blackbaud/skyux2/blob/master/skyux-spa-visual-tests/src/app/alert/alert-visual.component.html</a>
```

Comment:
@Blackbaud-BobbyEarl Can you share what the correct link should be?

Reply:
I can.  There should now be four links.  Using the alert as an example, the four links would be:

- https://github.com/blackbaud/skyux-indicators/blob/master/src/app/visual/alert/index.html
- https://github.com/blackbaud/skyux-indicators/blob/master/src/app/visual/alert/alert-visual.component.ts
- https://github.com/blackbaud/skyux-indicators/blob/master/src/app/visual/alert/alert-visual.component.html
- https://github.com/blackbaud/skyux-indicators/blob/master/e2e/alert.e2e-spec.ts

Also, use `href` instead of `stacheRouterLink`.  `stacheRouterLink` should only be used for links pointing to a route within the same SPA.